src/index.js

```diff
@@ -0,0 +1,176 @@
+'use strict'
+
+const base32Encode = require('base32-encode')
+const debug = require('debug')
+const log = debug('jsipns')
+log.error = debug('jsipns:error')
+
+const ipnsEntryProto = require('./pb/ipns.proto')
+const ERRORS = require('./errors')
+
+/**
+ * Creates a new ipns entry and signs it with the given private key.
+ * Note: This function does not embed the public key. If you want to do that, use `EmbedPublicKey`.
+ *
+ * @param {Object} privateKey private key for signing the record.
+ * @param {string} value value to be stored in the record.
+ * @param {number} seq sequence number of the record.
+ * @param {string} eol end of life datetime of the record.
+ * @param {function(Error)} [callback]
+ * @returns {Promise|void}
+ */
+const create = (privateKey, value, seq, eol, callback) => {
+  const validity = eol.toISOString()
+  const validityType = ipnsEntryProto.ValidityType.EOL
+
+  sign(privateKey, value, validityType, validity, (error, signature) => {
+    if (error) {
+      log.error(error)
+      return callback(error)
+    }
+
+    const entry = {
+      value: value,
+      signature: signature, // TODO confirm format compliance with go-ipfs
+      validityType: validityType,
+      validity: validity,
```

Comment:
I'm not sure how this relates to `ttl` in the protobuf definition - could you explain?

Reply:
This proto definition has to be the same as go, in order to guarantee the interoperability among them. 

The `ttl` parameter will also be used in the IPNS for caching purposes, as you can see [here](https://github.com/ipfs/go-ipfs/blob/master/core/commands/publish.go#L73). But since it is still experimental in go, I decided to leave it with low priority for now.